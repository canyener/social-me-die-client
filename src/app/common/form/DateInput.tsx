import { FieldRenderProps } from 'react-final-form'
import { Form, FormFieldProps, Label } from 'semantic-ui-react'
import { DateTimePicker } from 'react-widgets'

interface IProps
  extends FieldRenderProps<Date, HTMLElement>,
  FormFieldProps {
  id?: string
}

const DateInput: React.FC<IProps> = ({
  input,
  width,
  placeholder,
  date = false,
  time = false,
  meta: { touched, error },
  ...rest
}) => {
  return (
    <Form.Field error={touched && !!error} width={width}>
      <DateTimePicker
        placeholder={placeholder}
        value={input.value || null}
        onChange={input.onChange}
        date={date}
        time={time}
        {...rest}
      />
      {
        touched && error && (
          <Label basic color='red'>{error}</Label>
        )
      }
    </Form.Field>
  )
}

export default DateInput