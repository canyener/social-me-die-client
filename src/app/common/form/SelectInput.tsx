import { FieldRenderProps } from 'react-final-form'
import { Form, FormFieldProps, Label, Select } from 'semantic-ui-react'

interface IProps 
  extends FieldRenderProps<string, HTMLSelectElement>, 
    FormFieldProps { }

const SelectInput: React.FC<IProps> = ({
  input,
  width,
  options,
  placeholder,
  meta: { touched, error }
}) => {
  return (
    <Form.Field error={touched && !!error} width={width}>
      <Select 
        value={input.value}
        onChange={(_e, data) => input.onChange(data.value)}
        placeholder={placeholder}
        options={options}
      />
      {
        touched && error && (
          <Label basic color='red'>{error}</Label>
        )
      }
    </Form.Field>
  )
}

export default SelectInput