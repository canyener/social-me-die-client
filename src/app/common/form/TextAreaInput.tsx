import { FieldRenderProps } from "react-final-form"
import { Form, FormFieldProps, Label } from "semantic-ui-react"

interface IProps
  extends FieldRenderProps<string, HTMLTextAreaElement>,
  FormFieldProps { }

const TextAreaInput: React.FC<IProps> = ({
  input,
  width,
  rows,
  placeholder,
  meta: { touched, error }
}) => {
  return (
    <Form.Field error={touched && !!error} width={width}>
      <textarea rows={rows} {...input} placeholder={placeholder} /> 
      {
        touched && error && (
          <Label basic color='red'>
            {error}
          </Label>
        )
      }
    </Form.Field>
  )
}

export default TextAreaInput