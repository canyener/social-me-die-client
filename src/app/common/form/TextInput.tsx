import { FieldRenderProps } from "react-final-form"
import { Form, FormFieldProps, Label } from "semantic-ui-react"

interface IProps
  extends FieldRenderProps<string, HTMLInputElement>,
  FormFieldProps { }

const TextInput: React.FC<IProps> = ({
  input,
  width,
  type,
  placeholder,
  meta: { touched, error }
}) => {
  return (
    <Form.Field error={touched && !!error} width={width}>
      <input type={type} {...input} placeholder={placeholder} />
      {
        touched && error && (
          <Label basic color='red'>
            {error}
          </Label>
        )
      }
    </Form.Field>
  )
}

export default TextInput 