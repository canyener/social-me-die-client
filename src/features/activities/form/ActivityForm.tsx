import React from 'react'
import { Segment, Form, Button } from 'semantic-ui-react'

interface IProps {
  setEditMode: (editMode: boolean) => void
}

export const ActivityForm: React.FC<IProps> = ({ setEditMode }) => (
  <Segment clearing>
    <Form>
      <Form.Input placeholder='Title' />
      <Form.TextArea rows={2} placeholder='Description' />
      <Form.Input placeholder='Category' />
      <Form.Input type='date' placeholder='Date' />
      <Form.Input placeholder='City' />
      <Form.Input placeholder='Venue' />
      <Button floated='right' positive type='submit' content='Submit' />
      <Button onClick={() => setEditMode(false)} floated='right' type='button' content='Cancel' />
    </Form>
  </Segment>
)
