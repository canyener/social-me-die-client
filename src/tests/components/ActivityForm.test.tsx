import React from 'react'
import { shallow } from 'enzyme'

import { ActivityForm } from '../../features/activities/form/ActivityForm'

test('Should render ActivityForm correctly', () => {
  const wrapper = shallow(<ActivityForm setEditMode={() => {}} />)

  expect(wrapper).toMatchSnapshot()
})

test('Should call setEditMode with false when Cancel button is clicked ', () => {
  const setEditModeSpy = jest.fn()

  const wrapper = shallow(<ActivityForm setEditMode={setEditModeSpy} />)

  wrapper.find('Button').at(1).simulate('click')

  expect(setEditModeSpy).toHaveBeenCalledWith(false)
})