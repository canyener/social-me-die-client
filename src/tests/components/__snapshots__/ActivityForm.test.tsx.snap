// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ActivityForm correctly 1`] = `
<Segment
  clearing={true}
>
  <Form
    as="form"
  >
    <FormInput
      as={[Function]}
      control={[Function]}
      placeholder="Title"
    />
    <FormTextArea
      as={[Function]}
      control={[Function]}
      placeholder="Description"
      rows={2}
    />
    <FormInput
      as={[Function]}
      control={[Function]}
      placeholder="Category"
    />
    <FormInput
      as={[Function]}
      control={[Function]}
      placeholder="Date"
      type="date"
    />
    <FormInput
      as={[Function]}
      control={[Function]}
      placeholder="City"
    />
    <FormInput
      as={[Function]}
      control={[Function]}
      placeholder="Venue"
    />
    <Button
      as="button"
      content="Submit"
      floated="right"
      positive={true}
      type="submit"
    />
    <Button
      as="button"
      content="Cancel"
      floated="right"
      onClick={[Function]}
      type="button"
    />
  </Form>
</Segment>
`;
